### ISSUE TITLE
Pointer file displayed on GitHub instead of actual file

### ISSUE THREAD

**Issue by cmaumet** [github/git-lfs#1056]
We (with @AlexBowring) are using git-lfs in our public repository: https://github.com/incf-nidash/nidmresults-examples and for one of our commits (https://github.com/incf-nidash/nidmresults-examples/commit/f3dc7eaafd7f7d69ea673479e8311515e077c103), the files stored by git-lfs are displayed as pointers (instead of the actual file). 

An example of file is https://github.com/incf-nidash/nidmresults-examples/blob/f3dc7eaafd7f7d69ea673479e8311515e077c103/spm_t_test/RPV.nii showing up as 
```
version https://git-lfs.github.com/spec/v1
oid sha256:e88c0b9c2cb607d5684467c9143e2ba4c96ea7979a2d14175d4dd1122da32bfc
size 4743512
```
while in the full commit listing (https://github.com/incf-nidash/nidmresults-examples/commit/f3dc7eaafd7f7d69ea673479e8311515e077c103) this file is actually listed as "Git LFS file not shown". 

We are not experiencing this issue with other commits (e.g. https://github.com/incf-nidash/nidmresults-examples/commit/e92760f172d62cd80782d848afd2b55b76eb1ba3) and could not track down what the error was. 

Would you have any advice for us? Thank you!

**Comment by technoweenie** [comment]
It's because that object hasn't been pushed to the server. Try running `git lfs push --all` from your local repository to fix.

Also, this is a GitHub.com issue, and has nothing to do with this open source project. In the future, you can just hit me up at https://github.com/contact.

**Comment by cmaumet** [comment]
@technoweenie: I have described our issue at https://github.com/contact but we got no solution so far and it's getting worse... There are now non-pushed git-lfs files in the master branch which makes it impossible to clone the repository. To avoid spreading the issue further, we are on hold awaiting a fix. Could you let me know what is the best thing for me to do now? Thank you!

**Comment by technoweenie** [comment]
Sorry, clearly I was too quick to take this private. I found the bug in a command that Git LFS uses to scan the repository history:

```bash
$ git rev-list --objects --do-walk spm_t_test
fatal: ambiguous argument 'spm_t_test': both revision and filename
Use '--' to separate paths from revisions, like this:
'git <command> [<revision>...] -- [<file>...]'
```

I gave you a workaround in our support thread, which is basically just rename the branch and try again:

```bash
$ git checkout -b spm_t_test-pr
$ git lfs push --all origin spm_t_test-pr
```

Until we find a workaround with the `git rev-list` command, you'll want to make sure your branch names don't coincide with filenames. I don't currently know enough about the command to say that we'll be able to work around it. But, we will make the command tell you why it's failing, so you're not left scratching your head for two weeks.

**Comment by epriestley** [comment]
#1096 has a fix, but the quick version is that adding `--` at the end of the argument list (with no paths) disambiguates `spm_t_test` as a ref instead of a path.

**Comment by technoweenie** [comment]
Thanks for the fix, @epriestley :metal: I'm keeping this issue open until I can fix the bad error reporting. It's pretty terrible that a faulty command raises an obvious error that goes unreported to the end user.

**Comment by epriestley** [comment]
Yeah -- #1100 is probably a variant of this which can _only_ be fixed by fixing the error reporting.

**Comment by ttaylorr** [comment]
Looks like things should have been fixed up with #1096, so I'm going to close this one out for now.

If the problem shows up again, please make sure to comment on this issue!
